pkgs/development/compilers/ocaml/camlboot.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, enableParallelBuilding ? true
+, guile
+}:
+
+stdenv.mkDerivation rec {
+  name = "camlboot";
+
+  src = fetchFromGitHub {
+    owner = "ekdohibs";
+    repo = "camlboot";
+    rev = "b103dc52411d94c8df69bc8b7915dd43aa40f8e8";
+    sha256 = "JFj9C1HqHpUgbq6cFvIfd6mOPy67s2C9ZPWMqsrB4d0=";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [ guile ];
+
+  postPatch = ''
+    patchShebangs compile_ocamlc.sh
```

Comment:
If I add `runHook prePatch` and `runHook postPatch` at the beginning and the end of `postPatch` then the derivation gets stuck in a loop at build time.
I'm not familiar with hooks and I have no idea what is going on...

Reply:
Oof, this _is_ a postPatch. Do not add pre/postPatch here because that will cause an infinite loop as you discovered, sorry :facepalm: 

You do need to add preBuild/postBuild/preInstall/postInstall.

Reply:
oh, from your comment I assumed there was a difference between the "hook" and the "phase" `postPatch`, but I realize they are actually the same?

Reply:
I was mistaken, in my original suggestion I misread `patchPhase` instead of `postPatch`.

The phases are different from the pre/post hooks - the default phases call their pre/post hooks so you can add custom setup/cleanup logic with pre/post commands. But if you define a custom e.g. `buildPhase`, you have to explicitly say `runHook preBuild` at the start and `runHook postBuild`at the end or they won't get called. There's a nixpkgs-hammering linter check for this which probably explains it better: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md